.githooks/pre-push

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+npm run lint
```

Comment:
> I think we generally don't want the hook to run in those since we're expecting that the linter was already run if the changes went through a PR (or made it to master).

Double checking is not not necessarily a bad idea. 

in these couple of places do we know if `npm ci` was executed before the push ?

Reply:
Yeah, all of them are preceded by our [setup.yml workflow](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/ec7ee764948cbabaf6f640744a6b35d72203214d/.github/actions/setup/action.yml#L15), where `npm ci` is used

Reply:
- `.github/workflows/docs.yml` doesn't have `npm ci`

Reply:
It uses `./.github/actions/setup`, which does.

Reply:
I found references to `git push` in 
- `.github/workflows/docs.yml`
- `.github/workflows/release-cycle.yml`
- `.github/workflows/upgradeable.yml`

They all run 
```
      - name: Set up environment
        uses: ./.github/actions/setup
```
before doing the push, so they should have the npm environment set, and so `npm run lint` should no cause issue.

I'd rather keep it as a security measure of whatever we push is properly formated.

Reply:
My concern is less related to the lint failing, and more to the possibility of running arbitrary code along with `git push`, and that is easy to forget it.

I double-checked and seems everything is fine, but we'll need to be careful from now on whenever we add a `git push` to any CI workflow.